diff --git a/src/mame/sega/model2_v.cpp b/src/mame/sega/model2_v.cpp
--- a/src/mame/sega/model2_v.cpp
+++ b/src/mame/sega/model2_v.cpp
@@ -2013,6 +2013,15 @@ u32 *model2_state::geo_matrix_write(geo_state *geo, u32 opcode, u32 *input)
 	/* read in the transformation matrix */
 	for (i = 0; i < 12; i++)
 		geo->matrix[i] = u2f(*input++);
+	static unsigned vonj_geometry_matrix_trace_count;
+	if ((machine().system().name == std::string_view("vonj")) && (vonj_geometry_matrix_trace_count < 1024))
+	{
+		logerror("vonj_geometry_matrix: time=%f m=%f,%f,%f,%f,%f,%f,%f,%f,%f t=%f,%f,%f\n",
+			machine().time().as_double(), geo->matrix[0], geo->matrix[1], geo->matrix[2], geo->matrix[3],
+			geo->matrix[4], geo->matrix[5], geo->matrix[6], geo->matrix[7], geo->matrix[8],
+			geo->matrix[9], geo->matrix[10], geo->matrix[11]);
+		vonj_geometry_matrix_trace_count++;
+	}
 
 	return input;
